middlewares/auth.js

```diff
@@ -54,7 +54,7 @@ module.exports.isAuthV2 = async (req, res, next) => {
   try {
     const tokenPayload = await isAuthTokenValid(token, process.env.SECRET);
     const user = await UsersFunction.findUserById(User, tokenPayload.user_id);
-    UsersFunction.updateLastActiveAt(User, user?.user_id);
+    await UsersFunction.updateLastActiveAt(User, user?.user_id);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is good as it ensures that the `updateLastActiveAt` function call is awaited before proceeding, which can prevent potential race conditions or unexpected behavior due to the asynchronous nature of JavaScript. However, there's a potential issue: if `user` is null or undefined, calling `user?.user_id` will return undefined and this might not be the expected behavior for the `updateLastActiveAt` function. It would be better to check if the user exists before updating their last active timestamp.

```diff
-    await UsersFunction.updateLastActiveAt(User, user?.user_id);
+    if (user) {
+        await UsersFunction.updateLastActiveAt(User, user.user_id);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->